tests/test_core.py

```diff
@@ -1624,6 +1624,7 @@ def test_iostream_ctors(self):
 
   @also_with_standalone_wasm
   def test_ctors_no_main(self):
+    self.emcc_args.append('--no-entry')
```

Comment:
why is this needed? will users need to do it too?

Reply:
No, this is only needs to that the test still passed in STRICT mode.

Reply:
Eventually thought, this is the plan, that users would either need to pass an empty list of EXPORTED_FUNCTIONS of --no-entry (one or the other).   We agreed on this already I believe.

Reply:
This test doesn't pass `STRICT`, though?

Reply:
I'm asking, is this for future use, or was it needed now in a way that I'm missing?

Reply:
I've added a new mode below called `strict` that runs that entire suite in strict mode.  Like safe heap mode it doesn't run in CI but is useful for keeping out test suite clean.

Reply:
See below.

Reply:
Oh, if it's just for strict mode, how about an if on that? But if we want to do this by default later I guess that's pointless, so sgtm as is.